### ISSUE TITLE
Issue with Viridescent Venerer debuff duration for off field characters

### ISSUE THREAD

**Issue by TheHobosSon** [thisRaptori/genshin-rotation-app#37]
The elemental res debuff initiated by the Viridescent Venerer 4-piece set bonus only applies when the character equipping it causes a swirl and is on the field. For things that only swirl once it shows the debuff timer as the correct length but when it applies multiple swirls (e.g. Venti or Sucrose burst) it shows the duration of the debuff as ending 10 seconds after the duration of the ability, even if the character would be off the field when the ability ends and therefore when the final swirl happens, as seen in the image below.

![Viridescent Venerer](https://github.com/thisRaptori/genshin-rotation-app/assets/21371880/c7f6c1ca-c3c8-41b9-b2ea-a97b041051f3)

Additionally, since the elemental res debuff can't be triggered twice for the same element from the same source, if there are 2 characters using the set (I don't know why you would do that though) and Character B triggers a single swirl after the duration of Character A's ability applies it's final on-field swirl, the remaining duration of Character A's debuff should be overwritten.

![Viridescent Venerer Part 2](https://github.com/thisRaptori/genshin-rotation-app/assets/21371880/ea6b75ac-5df3-4a57-aca5-518ba5876dd0)

Also, just as a readability thing, in a scenario like the image below where a character triggers a single swirl (like from Sucrose Normal Attack), waits a bit into the debuff duration, and then swirls again, the debuff from the first swirl (Swirl A) would be overwritten so the marker for the end of it's duration is no longer applicable and makes the chart harder to read.

![Viridescent Venerer Part 3](https://github.com/thisRaptori/genshin-rotation-app/assets/21371880/f7da3c63-1975-4ac5-8b43-e2f7484844a0)

**Comment by thisRaptori** [comment]
Yeah definitely issues around those kinds of details, but unfortunately there isn't really any way to model them without essentially implementing the elemental system... I've got a rough idea of how that could be done, but have had no free time in recent months, so for now it'll just have to be treated as a rough guide 😕

**Comment by thisRaptori** [comment]
This should be fixed in the new beta version! The VV implementation now includes a check for whether the triggering character is on-field 🎉
